src/styles/blocks/mains.scss

```
@@ -0,0 +1,49 @@
+.calendar {
+  position: relative;
+  display: flex;
+  flex-wrap: wrap;
+
+  padding: 25px $calendarShihtPading 10px ;
+  gap: $gapForDays;
+  margin: auto;
+
+  width: $calendarMaxWidth;
+
+  @each $day in $days {
+    &--start-day-#{$day} > .calendar__day:first-child {
+      margin-left: $startDayShift;
+    }
+  }
+
+  @for $lastDay from 28 to 32 {
+    &--month-lenght-#{$lastDay} .calendar__day:nth-last-child(-n + 2) {
+      display: none;
+    }
+  }
+
+  &__day {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+
+    width: $daysWidth;
+    height: $daysHeight;
+
+    border: 1px solid black;
+    background-color: #eee;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translate(0, -20px);
+      transition: 0.5s;
+    }
+
+    @for $i from 1 to 32 {
```

```suggestion
    @for $index from 1 to 31 {
```
there are only 31 days (max) in a month :)
also, don't use short forms: ~$i~ -> ~$index~